expression/util.go

```diff
@@ -587,3 +587,20 @@ func GetIntFromConstant(ctx sessionctx.Context, value Expression) (int, bool, er
 	}
 	return intNum, false, nil
 }
+
+func CoveredBySchema(expr Expression, schema *Schema) bool {
```

Comment:
Seems it is equivalent to `ExprFromSchema`?

Reply:
Yeah, just noticed that. I've added a benchmark for both of them, `CoveredBySchema()` is one times faster than `ExprFromSchema()`:
```
BenchmarkCoveredBySchema-8              10000000               135 ns/op               0 B/op          0 allocs/op
BenchmarkCoveredBySchema-8              10000000               135 ns/op               0 B/op          0 allocs/op
BenchmarkCoveredBySchema-8              10000000               135 ns/op               0 B/op          0 allocs/op
BenchmarkCoveredBySchema-8              10000000               134 ns/op               0 B/op          0 allocs/op
BenchmarkCoveredBySchema-8              10000000               135 ns/op               0 B/op          0 allocs/op
BenchmarkExprFromSchema-8                5000000               304 ns/op             128 B/op          2 allocs/op
BenchmarkExprFromSchema-8                5000000               304 ns/op             128 B/op          2 allocs/op
BenchmarkExprFromSchema-8                5000000               303 ns/op             128 B/op          2 allocs/op
BenchmarkExprFromSchema-8                5000000               305 ns/op             128 B/op          2 allocs/op
BenchmarkExprFromSchema-8                5000000               302 ns/op             128 B/op          2 allocs/op
```

It seems that we can refactor `ExprFromSchema()` to `CoveredBySchema()`. The benchmark code is:
```go
func BenchmarkCoveredBySchema(b *testing.B) {
    conditions := []Expression{
        newFunction(ast.EQ, newColumn(0), newColumn(1)),
        newFunction(ast.EQ, newColumn(1), newColumn(2)),
        newFunction(ast.EQ, newColumn(2), newColumn(3)),
        newFunction(ast.EQ, newColumn(3), newLonglong(1)),
        newFunction(ast.LogicOr, newLonglong(1), newColumn(0)),
    }
    expr := ComposeCNFCondition(mock.NewContext(), conditions...)
    schema := &Schema{Columns: ExtractColumns(expr)}

    b.ResetTimer()
    for i := 0; i < b.N; i++ {
        CoveredBySchema(expr, schema)
    }
    b.ReportAllocs()
}

func BenchmarkExprFromSchema(b *testing.B) {
    conditions := []Expression{
        newFunction(ast.EQ, newColumn(0), newColumn(1)),
        newFunction(ast.EQ, newColumn(1), newColumn(2)),
        newFunction(ast.EQ, newColumn(2), newColumn(3)),
        newFunction(ast.EQ, newColumn(3), newLonglong(1)),
        newFunction(ast.LogicOr, newLonglong(1), newColumn(0)),
    }
    expr := ComposeCNFCondition(mock.NewContext(), conditions...)
    schema := &Schema{Columns: ExtractColumns(expr)}

    b.ResetTimer()
    for i := 0; i < b.N; i++ {
        ExprFromSchema(expr, schema)
    }
    b.ReportAllocs()
}
```